### PR TITLE
Add endpoint to track data useful for training draft models and heads

### PR DESCRIPTION
**Pull request by bwasti**
## Test Plan

Test a multi-GPU instance with proper env setup:

```
VLLM_USE_V1=1 VLLM_SERVER_DEV_MODE=1 LD_PRELOAD=/usr/local/cuda-12.4/lib64/libcublas.so python -m vllm.entrypoints.openai.api_server --model Qwen/Qwen3-1.7B --host 0.0.0.0 --port 8000 --tensor-parallel-size=2
```

Then hit the endpoints:

Enable
```
curl -X POST "http://localhost:8000/spec_decode/collection/start?output_dir=./eagle_data&hidden_state_layers=0,8,16,23"|jq
```
Prompt
```
curl http://localhost:8000/v1/completions -H "Content-Type: application/json" -d '{"model": "Qwen/Qwen3-1.7B", "prompt": "Once upon a time", "max_tokens": 50}' | jq
```
Check status:
```
curl http://localhost:8000/spec_decode/collection/status | jq
```

Then you can look in the numpy files dumped to to `./eagle_data/`:
```
python -c "import numpy as np; data = np.load('./eagle_data/spec_decode_data_rank0_000000.npz'); print(list(data.keys()))"
```

## Test Result

In the numpy files you'll get:
`['request_ids', 'timestamps', 'num_tokens', 'token_ids', 'token_lens', 'logits']`

The status endpoint:
```
$   curl http://localhost:8000/spec_decode/collection/status | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   455  100   455    0     0   148k      0 --:--:-- --:--:-- --:--:--  148k
{
  "enabled": true,
  "total_samples_collected": 100,
  "files_written": 1,
  "buffer_size": 0,
  "output_dir": "./eagle_data",
  "config": {
    "enabled": true,
    "output_dir": "./eagle_data",
    "hidden_state_layers": [
      0,
      8,
      16,
      23
    ],
    "samples_per_file": 1000,
    "max_buffer_size": 100,
    "collect_logits": true,
    "collect_hidden_states": true,
    "worker_rank": 0
  },
  "worker_rank": 0,
  "num_workers": 2,
  "workers": [
    {
      "rank": 0,
      "samples": 100,
      "files": 1,
      "buffer_size": 0
    },
    {
      "rank": 1,
      "samples": 100,
      "files": 1,
      "buffer_size": 0
    }
  ]
}
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [x] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Review by benchislett — COMMENTED** [review]
I don't really think we should maintain this feature. There are lots of metrics and data dumps that might be useful for training, and unless there's a burning integration I don't think it is worth maintaining them all
